### DISCUSSION TITLE
Loss function implementation

### DISCUSSION THREAD

**Discussion by kylerioux** [Project-MONAI/MONAI#4097] [Q&A]
Hello,

I am unsure of how the loss functions are implemented in MONAI. Will the cross entropy/dice/focal (any loss function) calculate the per-class loss and average the values, or is it NOT done on a per-class basis? 

I am aware that you can give per-class weights and 'skip background' - which makes me think the default behavior is to not average losses for each class. 

Any insight is appreciated :)

**Comment by Nic-Ma** [comment]
Hi @kylerioux ,

Thanks for your interest here.
I think it computes for every image and every class, then we support different reduction methods for the final value.
You can set as you expect:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/dice.py#L181

Thanks.
